### PR TITLE
Add body name 

### PR DESCRIPTION
**Pull request by maxfriedmann**
... which will produce the x-codegen-request-body-name extension in the root of the document

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: https://github.com/nestjs/swagger/issues/1212

## What is the new behavior?
`@ApiBody({name: "Foo"}) is working (again) as expected

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
- For us, coming from @nestjs/swagger@3.x it is not a new feature, it used to work like this. Nevertheless, for 4.x it could be a feature? 
- I checked "Docs have been added" since the current documentation is valid: It does not state that, during the migration, you would have lost this feature

### CODE REVIEW

File: lib/swagger-explorer.ts [hunk 1]
```diff
@@ -353,6 +353,8 @@ export class SwaggerExplorer {
       ...omit(requestBody, keysToRemove),
       ...this.mimetypeContentWrapper.wrap(consumes, pick(requestBody, 'schema'))
     };
+    if (requestBody.name)
+      document.root["x-codegen-request-body-name"] = requestBody.name;
```
Comment (kamilmysliwiec) [inline lib/swagger-explorer.ts:357]: Why is this needed?

Comment (maxfriedmann) [reply lib/swagger-explorer.ts:357]: If the property is not set, the default value will be used (the class name). see https://openapi-generator.tech/docs/swagger-codegen-migration/#body-parameter-name. With this change and the `@ApiBody({name: "..."})` which we still have in our code from @nestjs/swagger@3.x, our generated APIs look exactly like before.

Comment (maxfriedmann) [reply lib/swagger-explorer.ts:357]: Can I provide any additional information @kamilmysliwiec to support you in your merge decision? 

I know that the generators from openapi-generator.tech aren't the only api generators out there and `x-codegen-request-body-name` seems to be a non-standard for now. But it seems that others are picking up that property as well, see https://github.com/swagger-api/swagger-codegen-generators/pull/827/files

**Comment by kamilmysliwiec** [comment]
Thanks for your contribution! However, I can't merge this PR due to reasons outlined here https://github.com/nestjs/swagger/issues/1212#issuecomment-802841166
